ignite/metrics/epoch_metric.py

```diff
@@ -133,12 +138,37 @@ def compute(self) -> float:
             # Run compute_fn on zero rank only
             result = self.compute_fn(_prediction_tensor, _target_tensor)
 
+        # compute_fn outputs: scalars, tensors, tuple/list/mapping of tensors.
+        if not _is_scalar_or_collection_of_tensor(result):
+            raise TypeError(
+                "output not supported: compute_fn should return scalar, tensor, tuple/list/mapping of tensors"
+            )
+
         if ws > 1:
             # broadcast result to all processes
-            result = cast(float, idist.broadcast(result, src=0))
+            return apply_to_type(  # type: ignore
```

Comment:
I worked on this before, the Runtime errors is related to the broadcasting part. If you tried to print out anything while trying to handle tuple or list manually you will find out that it broadcasts the first tensor then it fails. I think what you need to figure out some way to broadcast more than just one tensor(tuples, lists, etc...) for solving this issue, I think then you will solve this issue. Good Luck!